### PR TITLE
 Support external PDF viewers using FileSaver.js

### PR DESCRIPTION
**Pull request by johnhawkinson**
Addresses https://github.com/freelawproject/recap/issues/183
This is high-priority for me. Can it make it into the 1.1.8 beta, please?

@plainsite can you please test this as well?

Here's how the options dialog looks now:

screen shot 2017-11-18 at 19 52 23

### CODE REVIEW

**Comment by johnhawkinson** [comment]
1f7d13a was after the dialog box mock was posted, so it's slightly improved.

**Comment by mlissner** [comment]
Just heard from somebody that uses FoxIt as their PDF reader on Windows 7 (FF57) that reported that they got a white screen of death whenever they tried to view PDFs. The fix was to enable the in-browser viewer. 

It'll be good to get this landed...I think it probably would have handled their issue. I think what I'd like to see, if you're willing, @johnhawkinson is:

1. Fixes/tweaks to the things mentioned in the initial review.
2. A tweak to do autodetection in Chrome, so that the option only appears/applies in FF.

I really don't like the addition of this option, but if we can make it a FF only thing, I think that's a good improvement.

File: content_delegate.js [hunk 1]
```diff
@@ -266,11 +269,14 @@ ContentDelegate.prototype.showPdfPage = function(
...
-    match[1] + '<iframe src="about:blank"' + match[3];
+    match[1] +
+    '<p>Waiting for download...<p><iframe src="about:blank"' +
+    match[3];
 
   // Download the file from the <iframe> URL.
   httpRequest(match[2], null, function (type, ab, xhr) {
-    console.info("RECAP: Successfully got PDF as arraybuffer via ajax request.");
+    console.info(
+      "RECAP: Successfully got PDF as arraybuffer via ajax request.");
```
[the remark this answers is not in the document]
Comment (johnhawkinson) [reply content_delegate.js]: Err, no? It cuts an 87-column line (which is more than 80) down to 70 columns. What would you rather see?

[the remark this answers is not in the document]
Comment (mlissner) [reply content_delegate.js]: I usually do:

    console.info("Some really long thing " + 
                 "the rest of it.");

Or:

    console.info("Something " + 
      "else");

But, obviously just a nitpick.

**Comment by mlissner** [comment]
> The fix was to enable the in-browser viewer.

NEWS: Their adblocker was blocking the uscourts.gov domain for unknown reasons. They're using Adblock Ultimate or something. Things seem to be working OK for them now.

File: options.html [hunk 2]
```diff
@@ -29,18 +29,20 @@
...
-  <fieldset><legend>PDF Filenames</legend>
+  <fieldset><legend>PDF Files</legend>
     <div>
+      <input type="checkbox" id="external_pdf">
+      <label>Force external PDF viewing (download PDFs)</label><br>
       <input type="radio" id="ia_style_filenames"
              name="filename_style" value="ia_style_filenames">
       <label for="ia_style_filenames">Use Internet Archive style (e.g.
-        gov.uscourts.cand.201881.46.0.pdf)</label>
+        gov.uscourts.cand.201881.46.0.pdf) naming</label>
```
[the remark this answers is not in the document]
Comment (johnhawkinson) [reply options.html]: Probably a little better :)

**Pull request merged**